code/modules/mob/living/basic/trooper/nanotrasen.dm

```diff
@@ -0,0 +1,114 @@
+	/// number of burst shots
+	var/burst_shots
+	/// Time between taking shots
+	var/ranged_cooldown = 1 SECONDS
+
+/mob/living/basic/trooper/nanotrasen/ranged/Initialize(mapload)
+	. = ..()
+	AddComponent(\
+		/datum/component/ranged_attacks,\
+		casing_type = casingtype,\
+		projectile_sound = projectilesound,\
+		cooldown_time = ranged_cooldown,\
+		burst_shots = burst_shots,\
+	)
+
+/mob/living/basic/trooper/nanotrasen/ranged/smg
+	ai_controller = /datum/ai_controller/basic_controller/trooper/ranged/burst
+	casingtype = /obj/item/ammo_casing/c46x30mm
+	projectilesound = 'sound/weapons/gun/smg/shot.ogg'
+	r_hand = /obj/item/gun/ballistic/automatic/wt550
+	burst_shots = 3
+	ranged_cooldown = 3 SECONDS
+
+/mob/living/basic/trooper/nanotrasen/ranged/assault
+	name = "Nanotrasen Assault Officer"
+	desc = "Nanotrasen Assault Officer. Contact CentCom if you saw him on your station. Prepare to die, if you've been found near Syndicate property."
+
+	casingtype = /obj/item/ammo_casing/a223/weak
+	burst_shots = 4
+	ranged_cooldown = 3 SECONDS
+	projectilesound = 'sound/weapons/gun/smg/shot.ogg'
+	r_hand = /obj/item/gun/ballistic/automatic/ar
+	loot = list(/obj/effect/mob_spawn/corpse/human/nanotrasenassaultsoldier)
+	mob_spawner = /obj/effect/mob_spawn/corpse/human/nanotrasenassaultsoldier
+
+/mob/living/basic/trooper/nanotrasen/ranged/elite
+	name = "Nanotrasen Elite Assault Officer"
+	desc = "Pray for your life, syndicate. Run while you can."
+	maxHealth = 150
+	health = 150
+	habitable_atmos = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_cold_damage = 0
+	casingtype = /obj/item/ammo_casing/energy/laser
+	burst_shots = 3
+	projectilesound = 'sound/weapons/laser.ogg'
+	ranged_cooldown = 5 SECONDS
+	faction = list(ROLE_DEATHSQUAD)
+	loot = list(/obj/effect/gibspawner/human)
+	mob_spawner = /obj/effect/mob_spawn/corpse/human/nanotrasenelitesoldier
+	r_hand = /obj/item/gun/energy/pulse/carbine/lethal
+
+/// A more peaceful variant that will only attack when attacked, or when another Nanotrasen officer calls for help.
+/mob/living/basic/trooper/nanotrasen/peaceful
+	desc = "An officer of Nanotrasen's private security force."
+	ai_controller = /datum/ai_controller/basic_controller/trooper/peaceful
+
+/mob/living/basic/trooper/nanotrasen/peaceful/Initialize(mapload)
+	. = ..()
+	var/datum/callback/retaliate_callback = CALLBACK(src, PROC_REF(ai_retaliate_behaviour))
+	AddComponent(/datum/component/ai_retaliate_advanced, retaliate_callback)
+
+/mob/living/basic/trooper/nanotrasen/peaceful/proc/ai_retaliate_behaviour(mob/living/attacker)
+	if (!istype(attacker))
+		return
+	for (var/mob/living/basic/trooper/nanotrasen/peaceful/potential_trooper in oview(src, 7))
+		potential_trooper.ai_controller.insert_blackboard_key_lazylist(BB_BASIC_MOB_RETALIATE_LIST, attacker)
+
+/mob/living/basic/trooper/nanotrasen/peaceful/ranged
```

Comment:
Does this need to be a subtype of `basic/trooper/nanotrasen/peaceful`?
It seems like it'd be easier to do it as a subtype of `basic/trooper/nanotrasen/ranged` and just swap the ai controller and add the element, rather than duplicating all the ranged vars.

Either way is a little sad in terms of replicating annoying inheritance patterns we are trying to remove from mobs but c'est la vie.